src/main/java/com/direwolf20/buildinggadgets/common/schema/template/Template.java

```diff
@@ -0,0 +1,241 @@
+package com.direwolf20.buildinggadgets.common.schema.template;
+
+import com.direwolf20.buildinggadgets.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.helpers.NBTHelper;
+import com.direwolf20.buildinggadgets.common.schema.BoundingBox;
+import it.unimi.dsi.fastutil.longs.Long2ObjectLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap.Entry;
+import it.unimi.dsi.fastutil.longs.LongArrayList;
+import it.unimi.dsi.fastutil.longs.LongList;
+import it.unimi.dsi.fastutil.objects.Object2IntLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import it.unimi.dsi.fastutil.objects.ObjectIterator;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.INBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockPos.Mutable;
+import net.minecraft.util.math.Vec3i;
+import net.minecraftforge.common.util.Constants.NBT;
+
+import java.util.*;
+
+public final class Template implements Iterable<TemplateData> {
+    private static final String KEY_POS = "pos";
+    private static final String KEY_TILE_POS = "tile_pos";
+    private static final String KEY_DATA = "data";
+    private static final String KEY_TILE_DATA = "tile_data";
+    private static final String KEY_AUTHOR = "author";
+    private static final int BYTE_MASK_12BIT = 0xF_FF;
+    private static final int BYTE_MASK_14BIT = 0x3F_FF;
+    private static final int BYTE_MASK_24BIT = 0xFF_FF_FF;
+    private static final long BYTE_MASK_40BIT = ((long) 0xFF_FF_FF_FF) << 8 | 0xFF;
+    private static final int MAX_NUMBER_OF_STATES = (1 << 24) - 1;
+    private final Long2ObjectMap<BlockState> states;
+    private final BoundingBox bounds;
+    private final String author;
+
+    public static Optional<Template> deserializeNBT(CompoundNBT nbt) {
+        if (! nbt.contains(KEY_POS, NBT.TAG_LONG_ARRAY) || ! nbt.contains(KEY_DATA, NBT.TAG_LIST))
+            return Optional.empty();
+
+        Long2ObjectMap<BlockState> states = new Long2ObjectLinkedOpenHashMap<>();
+        BoundingBox bounds = null;
```

Comment:
Redundant initializer

Reply:
This was done to make the point clearer,